Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -0,0 +1,259 @@
+      );
+      const mousePos = publicAPI.computeDisplayToWorld(
+        model.renderer,
+        callData.position.x,
+        callData.position.y,
+        test[2]
+      );
+      for (let i = 0; i < 3; i++) {
+        direction[i] =
+          model.widgetState.getHandle1List()[0].getOrigin()[i] - mousePos[i];
+      }
+      bv = 0;
+    } else {
+      const modifier = bv === 1 ? 1 : -1;
+      bv -= 1;
+      for (let i = 0; i < 3; i++)
+        direction[i] =
+          (model.widgetState.getHandle1List()[0].getOrigin()[i] -
+            model.widgetState.getHandle2List()[0].getOrigin()[i]) *
+          modifier;
+    }
+    model.representations[bv].getGlyph().setDirection(direction);
+  }
+
+  function setConeDirection() {
+    if (model.shapeHandle1 === 'cone') {
+      updateConeDirection(1);
+    }
+    if (model.shapeHandle2 === 'cone') {
+      updateConeDirection(2);
+    }
+  }
+
+  function setTextPosition(ofset, ofsetDir, linePosition) {
+    const vector = [0, 0, 0];
+    let oD = ofsetDir;
+    let lp = linePosition;
+    let os = ofset;
+    if (linePosition < 0 || linePosition > 1) {
+      console.log(
+        'Line position should always be between 0 and 1 to remain on the line'
+      );
+    } else lp = 1 - lp;
+    if (ofsetDir !== 0 && ofsetDir !== 1 && ofsetDir !== 2)
+      console.log('wrong ofset value');
+    if (oD === 1) oD *= -1;
+    if (oD === 0) os = 0;
+    for (let i = 0; i < 3; i++) {
+      vector[i] =
+        (model.widgetState.getHandle1List()[0].getOrigin()[i] -
+          model.widgetState.getHandle2List()[0].getOrigin()[i]) *
+          lp +
+        model.widgetState.getHandle2List()[0].getOrigin()[i];
+    }
+    vector[0] += os;
+    return vector;
+  }
+
+  function updateTextPosition() {
+    const obj = model.widgetState.getTextList()[0];
+    obj.setOrigin(
+      setTextPosition(model.offset, model.offsetDir, model.lineDir)
+    );
+  }
+
+  // --------------------------------------------------------------------------
+  // Left press: Select handle to drag
+  // --------------------------------------------------------------------------
+
+  publicAPI.handleLeftButtonPress = (e) => {
+    if (
+      !model.activeState ||
+      !model.activeState.getActive() ||
+      !model.pickable ||
+      ignoreKey(e)
+    ) {
+      return macro.VOID;
+    }
+
+    if (
+      model.activeState === model.widgetState.getMoveHandle() &&
+      model.widgetState.getHandle1List().length === 0
+    ) {
+      const moveHandle = model.widgetState.getMoveHandle();
+      const newHandle = model.widgetState.addHandle1();
+      newHandle.setOrigin(...moveHandle.getOrigin());
+      newHandle.setColor(moveHandle.getColor());
+      newHandle.setScale1(moveHandle.getScale1());
+    } else if (
+      model.activeState === model.widgetState.getMoveHandle() &&
+      model.widgetState.getHandle1List().length === 1 &&
+      model.widgetState.getHandle2List().length === 0
```

Comment:
Use explicit variable because you called some methods several times, and it can be more readable in else/if.